kubernetes/controller/internal/resolution/workerpool/event.go

```diff
@@ -0,0 +1,14 @@
+package workerpool
+
+// ResolutionEvent represents a notification that a component version resolution has completed.
+type ResolutionEvent struct {
```

Comment:
imo this should implement Unwrap and Error so it can be used like an error directly

Reply:
Uuu I like that idea.

Reply:
Actually simplified this because I thought it's overkill for what I'm using it.